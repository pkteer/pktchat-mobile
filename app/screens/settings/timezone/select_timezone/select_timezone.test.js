// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import SelectTimezone from './select_timezone';

describe('Settings SelectTimezone', () => {
    const baseProps = {
        onBack: jest.fn(),
        selectedTimezone: 'selected-timezone',
        initialScrollIndex: 1,
        timezones: [],
        theme: Preferences.THEMES.denim,
        isLandscape: false,
    };

    beforeEach(() => {
        jest.useFakeTimers();
    });

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(<SelectTimezone {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
