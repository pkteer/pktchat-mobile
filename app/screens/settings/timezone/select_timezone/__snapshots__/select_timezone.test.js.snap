// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings SelectTimezone should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  testID="settings.select_timezone.screen"
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "height": 38,
        "marginVertical": 5,
      }
    }
  >
    <Search
      autoCapitalize="none"
      backArrowSize={24}
      backgroundColor="transparent"
      blurOnSubmit={false}
      cancelTitle="Cancel"
      containerHeight={40}
      deleteIconSize={20}
      editable={true}
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(63,67,80,0.2)",
          "color": "#3f4350",
          "fontSize": 15,
        }
      }
      keyboardAppearance="light"
      keyboardShouldPersist={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(63,67,80,0.5)"
      returnKeyType="search"
      searchBarRightMargin={0}
      searchIconSize={24}
      selectionColor="rgba(63,67,80,0.5)"
      showArrow={false}
      showCancel={true}
      testID="settings.select_timezone.search_bar"
      tintColorDelete="rgba(63,67,80,0.5)"
      tintColorSearch="rgba(63,67,80,0.5)"
      titleCancelColor="#3f4350"
      value=""
    />
  </View>
  <FlatList
    data={Array []}
    getItemLayout={[Function]}
    initialScrollIndex={1}
    keyExtractor={[Function]}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="always"
    maxToRenderPerBatch={15}
    removeClippedSubviews={true}
    renderItem={[Function]}
    viewabilityConfig={
      Object {
        "itemVisiblePercentThreshold": 100,
        "waitForInteraction": false,
      }
    }
  />
</RNCSafeAreaView>
`;
